EIPS/eip-225.md

```diff
@@ -0,0 +1,474 @@
+   * Should be **`NONCE_DROP`** to propose deauthorizing `beneficiary` as a existing signer.
+   * Should be **`NONCE_AUTH`** to propose authorizing `beneficiary` as a new signer.
+   * **Must** be filled with zeroes on checkpoint (i.e. epoch transition) blocks.
+   * **Must** not take up any other value apart from the two above (for now).
+ * **`extraData`**: Combined field for signer vanity, checkpointing and signer signatures.
+   * First **`EXTRA_VANITY`** bytes (fixed) may contain arbitrary signer vanity data.
+   * Last **`EXTRA_SEAL`** bytes (fixed) is the signer's signature sealing the header.
+   * Checkpoint blocks **must** contain a list of signers (`N*20 bytes`) in between, **omitted** otherwise.
+   * The list of signers in checkpoint block extra-data sections **must** be sorted in ascending order.
+ * **`mixHash`**: Reserved for fork protection logic, similar to the extra-data during the DAO.
+   * **Must** be filled with zeroes during normal operation.
+ * **`ommersHash`**: **Must** be **`UNCLE_HASH`** as uncles are meaningless outside of PoW.
+ * **`timestamp`**: **Must** be at least the parent timestamp + **`BLOCK_PERIOD`**.
+ * **`difficulty`**: Contains the standalone score of the block to derive the quality of a chain.
+   * **Must** be **`DIFF_NOTURN`** if `BLOCK_NUMBER % SIGNER_COUNT != SIGNER_INDEX`
+   * **Must** be **`DIFF_INTURN`** if `BLOCK_NUMBER % SIGNER_COUNT == SIGNER_INDEX`
+
+### Authorizing a block
+
+To authorize a block for the network, the signer needs to sign the block's hash containing **everything except the signature itself**. The means that the hash contains every field of the header (`nonce` and `mixDigest` included), and also the  `extraData` with the exception of the 65 byte signature suffix. The fields are hashed in the order of their definition in the yellow paper.
+
+This hash is signed using the standard `secp256k1` curve, and the resulting 65 byte signature (`R`, `S`, `V`, where `V` is `0` or `1`) is embedded into the `extraData` as the trailing 65 byte suffix.
+
+To ensure malicious signers (loss of signing key) cannot wreck havoc in the network, each singer is allowed to sign **maximum one** out of **`SIGNER_LIMIT`** consecutive blocks. The order is not fixed, but in-turn signing weighs more (**`DIFF_INTURN`**) than out of turn one (**`DIFF_NOTURN`**).
+
+#### Authorization strategies
+
+As long as signers conform to the above specs, they can authorize and distribute blocks as they see fit. The following suggested strategy will however reduce network traffic and small forks, so it's a suggested feature:
+
+ * If a signer is allowed to sign a block (is on the authorized list and didn't sign recently).
+   * Calculate the optimal signing time of the next block (parent + **`BLOCK_PERIOD`**).
+   * If the signer is in-turn, wait for the exact time to arrive, sign and broadcast immediately.
+   * If the signer is out-of-turn, delay signing by `rand(SIGNER_COUNT * 500ms)`.
```

Comment:
Observations on Görli show that we have way too many reorganizations due to the out-of-turn delay often being lower than the network latency. I would strongly encourage to introduce something like a **`MIN_WAIT`** of `5000ms`, so that the out-of-turn delay is something around `MIN_WAIT + rand(SIGNER_COUNT * 500ms)`.

Reply:
Agreed, with just the random delay out of turn blocks may be published at essentially the same time as the in-turn block.  I'd highly recommend the minimum wait time and the 500ms multiplier should both be configurable like block period tends to be. The best values for them are highly affected by network latency and the chosen block period.

Reply:
I will argue that NO_TURN blocks should probably wait for at least half period , to give person inturn an chance.

Reply:
It's not affecting consensus, so clients could just go ahead and implement it.

Reply:
@5chdn I will make the change in parity for this.

Reply:
Any delay will make block times jump around so instead of 15 seconds, some blocks will take 22 seconds. That seems overly excessive to me. 

Also, if a signer is offline, there will be more than 1 gap, (the signer that siges out-of-turn next has a high probability of missing its turn because it cannot yet sign).

Reply:
I disagree, in an POA environment, the node are much more stable, i would argue that the distribution of block time is not Normial distributed or Bi Modal distributed, let's say some comptatnt node can generate inturn block and distributed at an relaibility of 99.9% (or 99% if super unlucky) ,  the NOTURN block should really just be 0.1%. 

Suppose we put NOTURN and INTURN block both together with no delay, then 99.9% of the time the network will reorg to INTURN block, which isn't ideal.

Reply:
> Any delay will make block times jump around so instead of 15 seconds, some blocks will take 22 seconds. That seems overly excessive to me.

Just like on mainnet :smiley: -- I don't see why this should be an issue, especially when this will drastically improve network stability.

Reply:
> > Any delay will make block times jump around so instead of 15 seconds, some blocks will take 22 seconds. That seems overly excessive to me.
> 
> Just like on mainnet 😃 -- I don't see why this should be an issue, especially when this will drastically improve network stability.

@5chdn  you  are saying you agree that more "potential" delay is better than frequent 1block reorg, righ?